source/common/http/http2/codec_impl.cc

```diff
@@ -447,7 +448,9 @@ void ConnectionImpl::StreamImpl::readDisable(bool disable) {
     --read_disable_count_;
     if (!buffersOverrun()) {
       scheduleProcessingOfBufferedData();
-      grantPeerAdditionalStreamWindow();
+      if (shouldAllowPeerAdditionalStreamWindow()) {
```

Comment:
This is probably obvious, but it's not quite clear to me how this change relates to the rest of this PR? Can you explain? (I mean, it *looks* like a fine change to not grant additional window unless we should, but it makes me wonder why we were not doing that previously)

Reply:
Ah, thanks for the explanation. That makes sense. I would probably fork it into a different PR, but I think it's also fine to leave here, just please make a note about it in the PR description. I presume we have test coverage of this path now, right?